XRTK-Core/Packages/com.xrtk.core/Runtime/Extensions/TypeExtensions.cs

```diff
@@ -12,22 +15,60 @@ namespace XRTK.Extensions
     /// </summary>
     public static class TypeExtensions
     {
-        private static readonly Dictionary<string, Type> TypeMap = new Dictionary<string, Type>();
+        private static readonly Dictionary<Guid, Type> TypeCache = new Dictionary<Guid, Type>();
+
+        private static IEnumerable<Type> allTypes = null;
+
+        private static IEnumerable<Type> AllTypes => allTypes ?? (allTypes = AppDomain.CurrentDomain.GetAssemblies()
+            .SelectMany(assembly => assembly.GetTypes())
+            .Where(type => type.IsClass && !type.IsAbstract));
 
         /// <summary>
-        /// Attempts to resolve the class reference by the <see cref="Type.AssemblyQualifiedName"/>.
+        /// Attempts to resolve the type using a the <see cref="System.Type.AssemblyQualifiedName"/> or <see cref="System.Type.GUID"/> as <see cref="string"/>.
         /// </summary>
-        /// <param name="classRef">The <see cref="Type.AssemblyQualifiedName"/> of the type to get.</param>
-        /// <returns>The <see cref="Type"/> from <see cref="classRef"/></returns>
-        public static Type ResolveType(string classRef)
+        /// <param name="typeRef">The <see cref="System.Type.GUID"/> or <see cref="System.Type.AssemblyQualifiedName"/> as <see cref="string"/>.</param>
+        /// <param name="resolvedType"></param>
+        /// <returns>The resolved <see cref="Type"/></returns>
+        public static bool TryResolveType(string typeRef, out Type resolvedType)
```

Comment:
Can we also have an overload that takes a GUID as a type instead of a string, so we can avoid string type conversions where we already have a GUID.
(updating this function to also call the GUID version)

Reply:
No, because we'd break backwards compatibility with `AssemblyQualifiedName`

Reply:
I'll add the overload